bench/wtperf/wtperf.c

```diff
@@ -707,24 +718,103 @@ worker(void *arg)
                  * Copy as much of the previous value as is safe, and be sure to NUL-terminate.
                  */
                 strncpy(value_buf, value, opts->value_sz_max - 1);
-                if (workload->update_delta != 0)
-                    update_value_delta(thread);
+
+                if (*op == WORKER_MODIFY)
+                    delta = workload->modify_delta;
+                else
+                    delta = workload->update_delta;
+
+                if (delta != 0)
+                    update_value_delta(thread, delta);
                 if (value_buf[0] == 'a')
                     value_buf[0] = 'b';
                 else
                     value_buf[0] = 'a';
                 if (opts->random_value)
-                    randomize_value(thread, value_buf);
-                cursor->set_value(cursor, value_buf);
-                if ((ret = cursor->update(cursor)) == 0)
-                    break;
+                    randomize_value(thread, value_buf, delta);
+
+                if (*op == WORKER_UPDATE) {
+                    cursor->set_value(cursor, value_buf);
+                    if ((ret = cursor->update(cursor)) == 0)
+                        break;
+                    goto op_err;
+                }
+
+                /*
+                 * Distribute the modifications across the whole document. We randomly choose up to
+                 * the maximum number of modifications and modify up to the maximum percent of the
+                 * record size.
+                 */
+                nmodify = MAX_MODIFY_NUM;
+
+                /*
+                 * The maximum that will be modified is a fixed percentage of the total record size.
+                 */
+                total_modify_size = strlen(value) / MAX_MODIFY_PCT;
```

Comment:
I think we should pre-calculate strlen(value) into a local variable. It is called 4 times (or 5 times if you include strlen(value_buf) which should be identical), which may be significant if the value buffer is configured to be very large.

Reply:
Fixed with a local variable for strlen(value), I left strlen(value_buf) as it is, because it is possible that value and value_buf sizes may differ.